packages/discovery-react-components/src/components/DiscoverySearch/types.ts

```diff
@@ -17,6 +18,14 @@ export interface DocumentProviderProps
   extracted_metadata?: JsonObject;
 }
 
+/**
+ * Document data for PDFSource
+ */
+export type PDFSourceDocument = {
+  __type: 'pdf';
+  source: PDFSource;
+};
```

Comment:
Instead of simply allowing `PDFSource` object, use an object with `__type` so that we'd be able to support various type of documents in future.